### PR TITLE
feat: floe labeling

### PR DESCRIPTION
**Pull request by cpaniaguam**
@danielmwatkins @ellenbuckley See `notebooks/track-floes/track-floes.ipynb` for a complete sample workflow with test data.

TODO: Update `track` in `https://github.com/WilhelmusLab/ice-floe-tracker-pipeline/blob/main/src/tracker.jl` so it does floe labeling.

#339

### CODE REVIEW

File: src/tracker/tracker.jl [hunk 1]
```diff
@@ -107,3 +107,47 @@ function pairfloes(
...
+    end
+end
+
+"""
+    tagfloes!(floedata, pairs)
+
+Tag floes in the `data.props` by assigning a unique `Floe_ID`` to each floe in each image using pair data in `pairs`. The ID is stored in the `Floe_ID` column of the `props` DataFrame in `data`.
+
+# Arguments
+- `data::Data`: the data structure containing the images and properties of the floes (output of `IFTPipeline.extractfeatures`). The `props` field of `data` is modified in-place.
```
Comment (tdivoll) [inline src/tracker/tracker.jl:132]:
```suggestion
- `floedata::Data`: the data structure containing the images and properties of the floes (output of `IFTPipeline.extractfeatures`). The `props` field of `data` is modified in-place.
```

File: src/tracker/tracker.jl [hunk 2]
```diff
@@ -107,3 +107,47 @@ function pairfloes(
...
+        return df[row, value_column]
+    else
+        return missing
+    end
+end
+
+"""
+    tagfloes!(floedata, pairs)
+
+Tag floes in the `data.props` by assigning a unique `Floe_ID`` to each floe in each image using pair data in `pairs`. The ID is stored in the `Floe_ID` column of the `props` DataFrame in `data`.
```
Comment (tdivoll) [inline src/tracker/tracker.jl:129]:
```suggestion
Tag floes in the `data.props` by assigning a unique `Floe_ID` to each floe in each image using pair data in `pairs`. The ID is stored in the `Floe_ID` column of the `props` DataFrame in `data`.
```

File: src/tracker/tracker.jl [hunk 3]
```diff
@@ -107,3 +107,47 @@ function pairfloes(
...
+
+# Arguments
+- `data::Data`: the data structure containing the images and properties of the floes (output of `IFTPipeline.extractfeatures`). The `props` field of `data` is modified in-place.
+- `pairs`: the output of `pairfloes`.
+"""
+function tagfloes!(props, pairs)
```
Comment (tdivoll) [inline src/tracker/tracker.jl:135]: Here the first arg is `props`, but in the docstring it is `floedata`. Maybe update the docstring so it is consistent?

**Review by tdivoll — APPROVED** [review]
This looks good so far. I just added a comment about updating the docstring so that the arguments match the names used in function itself.

**Comment by cpaniaguam** [comment]
Using a different implementation using early tagging and mappings for relabeling.

**Pull request merged**